### DISCUSSION TITLE
Standards components as nanopublications - worth pursuing?

### DISCUSSION THREAD

**Discussion by indiebio** [omigroup/omigroup#504] [Ideas]
Challenge: The way that we as a wider collective (beyond OMI) create standards is not ideal, not truly open, and difficult to interact with. There is a lot of duplication.

Idea: Instead of having a big "unit" of a standard

- our standards in question: https://github.com/omigroup/gltf-extensions/pulls (not sure if this is the best link. The reference link is: https://github.com/omigroup/gltf-extensions/pull/231)

The background of this idea is inspired by a video by the [Open Science Foundation](https://osf.io/) - particularly the open process approach. The video is about the contrast of open vs closed processes, it is relevant to academic publishing and scientific trust, but I think it applies to metaverse standards too: [Brian Nosek, 19minutes long](https://youtu.be/cZCvR37dzzA?si=_P-Vjc_Yu1aX3CAF)
11:36 min - the trustworthiness of research (and I would argue the same is true for standards) is more about the process than the outcome
14:00 min - the current closed model of evaluation and publication (and this is even for open-access papers) which I would argue is similar for current standards procedures:
 -- opague, a glimpse in time (the output is a paper, not the whole journey to get there), dichotomous (either it is published or not) and final.
The alternative, using an open process model (15:47 min): 
 -- transparent, showing the lifecycle, having many evaluations, versioned.
I think we can argue that we at OMI follow this approach already, to a large extent. Our challenge is how to integrate this with work that e.g. Khronos is doing, and we are also struggling with getting work that we did approved in an transparent and accountable way. 

I propose that we consider publishing components of the standards, in as small units as we can (starting with the easy ones) in a [Nanopublishing](https://nanopub.net/) format. I hypothesise that if this approach is followed, the smaller components can be approved more easily, and the smaller components can interface with smaller components from other standards organisations.

The discussion then raised a point about that some components of the standards are closely related and should stay together as a unit. While we can start with the simpler components to test the approach, the point of linked data is that the relationship is maintained, and can be illustrated through graphs, see this [graph](https://scholia.toolforge.org/topic/Q2632041#coauthor-graph) for example.

not sure if relevant: simple semantic domain modelling - SDML - is now in Debian testing: [https://tracker.debian.org/pkg/sdml](https://tracker.debian.org/pkg/sdml)

**Comment by indiebio** [comment]
**How this relates to my research (maybe)** 
I don't know much about any of this, so feel free to critique heavily :)  Even if we discard this as an idea hopefully it leads to some inspiration. This is relevant to me personally through my interest in how we facilitate the accountable transformation of expertise, and how we build new knowledge alliances.

**Reply by indiebio** [reply]
A meta observation after a discussion with a friend: I guess we are then similar to OSF and Debian in that we are trying to prove, or lead by example, that we do not need corporate institutions to establish standards / build the metaverse - or the internet: as Debian is to software, we are trying to find something for the wider infrastructure.

**Reply by indiebio** [reply]
how can we track, if that is the right word, granular bits of expertise transformation

**Comment by indiebio** [comment]
**More about nanopub** (from discussion with a friend): Nanopublishing is essentially a compact "scientific paper" containing only metadata.
Example:
...so that you can store "gosh, I found an interesting stone" while walking in the mountains, and then when you get home you can write a (conventional) paper where you reference stones #317, #4409 and #32 using conventional academic source reference system.
He was not convinced about the usefulness about nanopub for OMI because he said nanopub is about ownership of fractional ideas, and I was wondering if it (or something like it) be repurposed for identification of fractional components, that can then be recombined in ways that suit different groups (e.g. OMI vs Khronos). I wondered if a github PR could be a "stone"?
He responded that yes, you might (ab)use it for a PR but all that (in my understanding) provides is adding academic stickers to the PR, so that from an academic PoV you can more reliably resolve "this PR was done by indie-with-that-ORCID", and he has difficulty imagining how that added hinting provides any benefits for composing/evolving a spec.
Nanopub is a (data format and) platform for publishing radically tiny academic assets consisting of only links to externalized (often larger) assets

**Comment by indiebio** [comment]
**The thinking behind this**
So I see this as something like: This spec consists of these elements, #317, #4409 and #32. And then if we dive deeper, maybe elements #317, #4409 was authored by the glTF group (OMI) and #32 was authored by Khronos. 
I think at its core we are trying to find an alternative to this multiple standards thing: https://xkcd.com/927/, by breaking up the standard into nano pieces (I think)
So that the bits where we agree builds common ground, and then the divergent bits we can ... do something else with, or say it's for this separate path if you do X, Y or Z or something

**Comment by indiebio** [comment]
Then a thread for the **academic link**. I think we wanted DOI's for the PRs that is wider than our github? (Why?)
The academic link came in because we think there is novelty in how we approach this, in the open process way, versus the more closed, corporate, traditional way that Khronos approaches it, which I think has similarity with academic publishing.

**Reply by fire** [reply]
We can apply to be a DOI publisher; it's not impossible. With the publisher status, any document we generate will have a doi.

**Reply by fire** [reply]
https://www.doi.org/the-community/what-are-registration-agencies/

**Reply by fire** [reply]
We can also store 50GB per doi with Zendo https://cassgvp.github.io/github-for-collaborative-documentation/docs/tut/6-Zenodo-integration.html

**Reply by indiebio** [reply]
Sure. My question is, I missed that part of the meeting, WHY do we want the DOIs?

**Reply by fire** [reply]
I want our standards to be more permanent. A doi is a standard way of registering ids to documents.

**Reply by fire** [reply]
Like a web url but presumably more stable.

**Comment by indiebio** [comment]
**Does this link to the challenge of getting PRs approved?**
When the glTF group worked on a PR, then it needs to get approved, in a transparent way. But we're small, there's not many members, so it gets to a point where aaron has to approve the PR he worked on himself. So I think part of this conversation was about building the story so other people understand it and can approve it, or if not, that the story is transparent and can be queried even if aaron approved it himself. That's how I understand it, anyway

**Comment by indiebio** [comment]
**More about Open Science Foundation (OSF)**
it is like sourceforge or github or gitlab (e.g. salsa for Debian) but for science data and publications. essentially, a public library that used DOIs as references, with an emphasis on public, compared to its competitors, which includes commercial academic publishers. 
So I would say we are inspired by, but not intending to join, or perhaps even duplicate, OSF
OSF is a platform for publishing conventional academic assets

**Comment by indiebio** [comment]
We need to try this with an **example**, what would be a good one? Small, modular, perhaps that had the Khronos tension?
